EKF/control.cpp

```diff
@@ -787,7 +787,9 @@ void Ekf::controlHeightSensorTimeouts()
 	const bool continuous_bad_accel_hgt = isTimedOut(_time_good_vert_accel, (uint64_t)_params.bad_acc_reset_delay_us);
 
 	// check if height has been inertial deadreckoning for too long
-	const bool hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6);
+	// in vision hgt mode check for vision data
+	const bool hgt_fusion_timeout = isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6)  ||
+			 (_control_status.flags.ev_hgt && !isRecent(_time_last_ext_vision, 5 * EV_MAX_INTERVAL));
```

Comment:
IMO, ev_hgt fusion should simply update `_time_last_hgt_fuse` as it is also another source of height fusion.

Reply:
I tried that, but I could not get any vision timeouts then

Reply:
I think the difference is that `isTimedOut(_time_last_hgt_fuse, (uint64_t)5e6)` is a timeout of 5 seconds while `5 * EV_MAX_INTERVAL` is 1 second. Is that the issue? Do you just need it to be more reactive?

Reply:
This is definitely the case - but to be honest, I think I did not get a timeout at all, that's also why I played around with `_time_last_hgt_fuse` a bit.

Reply:
I think I found why this was needed. There is a bug causing height fusion to not stop correctly (`_time_last_hgt_fuse` keeps getting updated long after data stops)

Reply:
There was no check for `_ev_data_ready` before fusing the height meas, leading to a stale measurement being fused continuously.